Automattic/wp-calypso
# Calypso Build Package: Make `WP` flag add JSX -> `@wordpress/element` transforms

Per Jon's suggestion.

#### Changes proposed in this Pull Request

* Turn calypso-build's wordpress-element Babel preset into a function
  * I was getting `Plugin/Preset files are not allowed to export objects, only functions` :man_shrugging: 
* Change `calypso-build/babel.config.js` into an actual Babel preset, `calypso-build/babel/default.js`. This allows our webpack config to pass a list of presets (whose elements depend on the `WP` flag) to the Babel loader, without needing to worry if they're going to collide with a Babel config.
* Calypso Build Package: Make `WP` flag add JSX -> `@wordpress/element` transforms

#### Testing instructions

Do the following both on `master`, and this branch:

```
npm run distclean && npm ci
npx lerna run build
```

Keep the resulting `apps/*/dist/` directories, and `diff` them. They should be identical.

Furthermore, verify that Calypso still builds and runs (`npm start`).
Finally, verify that it's still possible to build the client in production mode, for the `fallback` target: `npm run build-client-fallback`.
